src/main/java/lux/fm/bookingservice/service/impl/BookingServiceImpl.java

```diff
@@ -1,47 +1,103 @@
 import lux.fm.bookingservice.repository.booking.BookingRepository;
+import lux.fm.bookingservice.repository.user.UserRepository;
 import lux.fm.bookingservice.service.BookingService;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
 public class BookingServiceImpl implements BookingService {
     private final BookingRepository bookingRepository;
     private final BookingMapper bookingMapper;
+    private final AccommodationRepository accommodationRepository;
+    private final UserRepository userRepository;
 
     @Override
-    public List<BookingResponseDto> findBookingsByUserIdAndStatus(
-            Long userId,
-            Status status) {
+    public List<BookingResponseDto> findBookingsByUserIdAndStatus(Long userId, Status status) {
         return bookingRepository.findByUserIdAndStatus(userId, status)
-                .stream()
-                .map(bookingMapper::toDto)
+                .stream().map(bookingMapper::toDto)
                 .toList();
     }
 
     @Override
     public List<BookingResponseDto> findMyBookings(String username) {
-        return bookingRepository.findBookingsByUserEmail(username)
-                .stream()
+        return bookingRepository.findBookingsByUserEmail(username).stream()
                 .map(bookingMapper::toDto)
                 .toList();
     }
 
     @Override
     public BookingResponseDto findBookingById(String username, Long id) {
-        List<Booking> bookings =
-                bookingRepository.findBookingsByUserEmail(username);
-        return bookings.stream()
-                .filter(e -> e.getId().equals(id))
+        List<Booking> bookings = bookingRepository.findBookingsByUserEmail(username);
+        return bookings.stream().filter(e -> e.getId().equals(id))
                 .map(bookingMapper::toDto)
                 .findFirst()
                 .get();
     }
+
+    @Override
+    public BookingResponseDto addBooking(BookingRequestDto request) {
+        validate(request);
+
+        User user = getCurrentlyAuthenticatedUser();
+        Booking booking = new Booking();
+        booking.setUser(user);
+        booking.setAccommodation(getAccommodation(request.accommodationId()));
+        booking.setCheckIn(request.checkIn());
+        booking.setCheckOut(request.checkOut());
+        booking.setStatus(Status.PENDING);
+        return bookingMapper.toDto(bookingRepository.save(booking));
+    }
+
+    private User getCurrentlyAuthenticatedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        return userRepository.findByEmail(userDetails.getUsername())
+                .get();
```

Comment:
authentication.getUsernmae()
```suggestion
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        return userRepository.findByEmail(authentication .getUsername());
```